cmd/kubeadm/app/phases/addons/dns/dns.go

```diff
@@ -48,6 +49,28 @@ const (
 	kubeDNSFederation          = "federations"
 )
 
+// DeployedDNSAddon returns the type of DNS addon currently deployed
+func DeployedDNSAddon(client clientset.Interface) (string, string, error) {
+	deploymentsClient := client.AppsV1().Deployments(api.NamespaceSystem)
+	deployments, err := deploymentsClient.List(metav1.ListOptions{LabelSelector: "k8s-app=kube-dns"})
```

Comment:
So long as we always used this label even for previous CoreDNS deploys, this is ok.

Reply:
Yes, this label has been used consistently within kubeadm and this is also consistent with the use in /cluster for use with the dns pod autoscaler used there.